runtime/doc/api-functions.txt

```diff
@@ -0,0 +1,722 @@
+*api-funcs.txt*         Neovim API Function Reference                   {Nvim}
+
+Note: This documentation is generated from Neovim's API source code.
+
+Contents:
+
+1. Buffer Functions   |nvim-api-buffer|
+2. Tabpage Functions  |nvim-api-tabpage|
+3. UI Functions       |nvim-api-ui|
+4. Vim Functions      |nvim-api-vim|
+5. Window Functions   |nvim-api-window|
+
+==============================================================================
+1. Buffer Functions                                          *nvim-api-buffer*
+
+nvim_buf_line_count({buffer})                          *nvim_buf_line_count()*
+                Gets the buffer line count
+
+                Parameters:~
+                    {buffer}  Buffer handle
+
+                Return:~
+                    Line count
+
+                                                        *nvim_buf_get_lines()*
+nvim_buf_get_lines({channel_id}, {buffer}, {start}, {end},
```

Comment:
The safest way would actually be to get the argument list from the metadata (available in `build/funcs_data.mpack`). This will also make it possible to expose more info like if the function is available in vimL or not. I guess this could be done in a follow PR, to not block this already very useful PR.